app/(dashboard)/dashboard/interviews/_components/ExportInterviewsDialog.tsx

```diff
@@ -0,0 +1,137 @@
+import type { Interview } from '@prisma/client';
+import { useState, type Dispatch, type SetStateAction } from 'react';
+import { Button } from '~/components/ui/Button';
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from '~/components/ui/dialog';
+import { useToast } from '~/components/ui/use-toast';
+import { exportSessions } from '../_actions/export';
+import ExportOptionsView from './ExportOptionsView';
+import ExportingStateAnimation from './ExportingStateAnimation';
+
+export type ExportOptions = typeof defaultExportOptions;
+
+type ExportInterviewsDialogProps = {
+  open: boolean;
+  setOpen: Dispatch<SetStateAction<boolean>>;
+  interviewsToExport: Interview[];
+  setInterviewsToExport: Dispatch<SetStateAction<Interview[] | undefined>>;
+};
+
+const defaultExportOptions = {
+  exportGraphML: true,
+  exportCSV: true,
+  globalOptions: {
+    exportFilename: `networkCanvasExport-${Date.now()}`,
+    unifyNetworks: false,
+    useScreenLayoutCoordinates: false,
+  },
+};
+
+const setOptionsToLocalStorage = (options: ExportOptions) => {
+  localStorage.setItem('exportOptions', JSON.stringify(options));
+};
+
+const getLocalExportOptions = () => {
+  const localExportOptions = localStorage.getItem('exportOptions');
+  return localExportOptions
+    ? (JSON.parse(localExportOptions) as ExportOptions)
+    : null;
+};
+
+export const ExportInterviewsDialog = ({
+  open,
+  setOpen,
+  interviewsToExport,
+  setInterviewsToExport,
+}: ExportInterviewsDialogProps) => {
+  const { toast } = useToast();
+  const [isExporting, setIsExporting] = useState(false);
+  const [exportOptions, setExportOptions] = useState(
+    getLocalExportOptions() ?? defaultExportOptions,
+  );
+
+  const handleConfirm = async () => {
+    // start export process
+    setIsExporting(true);
+    try {
+      const interviewIds = interviewsToExport.map((interview) => ({
+        id: interview.id,
+      }));
+
+      const result = await exportSessions(interviewIds, exportOptions);
+      handleCloseDialog();
+
+      if (result.data) {
+        const link = document.createElement('a');
+        link.href = result.data.url;
+        link.download = result.data.name; // Zip filename
+        link.click();
```

Comment:
Let's abstract this into a custom hook. One might already exist. Please update other instances where we do this kind of thing (export participants?).

Reply:
This should ideally support passing either a URL, OR creating a data URL from a base64 encoded string of some kind.

Reply:
This one is optional.